pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -227,6 +235,29 @@ func (rc *reconciler) reconcile() {
 			// If timeout is true, skip verifySafeToDetach check
 			glog.V(5).Infof(attachedVolume.GenerateMsgDetailed("Starting attacherDetacher.DetachVolume", ""))
 			verifySafeToDetach := !timeout
+			if utilfeature.DefaultFeatureGate.Enabled(features.NodeShutdown) {
```

Comment:
This logic probably should not be in the reconciler. The reconciler is intended to be a dumb loop that executes every 100ms and  simply diffs the actual state with the desired state and triggers attach/detach operations if they don't match. This will cause the node object to be fetched repeatedly in this critical path.

Can we can move this logic to the attach/detach "populator" and modify the desired state cache to keep track of if a node shutdown is in progress. If you go down that path, very careful about keeping that bit in sync with ground truth (we **never** want to pull disks without a grace period since that can result in data corruption, which will happen if there is a bug in this logic).